packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/ModelProvider.cs

```diff
@@ -235,14 +235,46 @@ private ConstructorProvider BuildFullConstructor()
 
         private ValueExpression GetExpression(ParameterProvider parameter)
         {
-            if (parameter.Property is not null && parameter.Property.IsDiscriminator)
+            if (parameter.Property is not null && parameter.Property.IsDiscriminator && _inputModel.DiscriminatorValue != null)
```

Comment:
Can IsDiscriminator be true and DiscriminatorValue be null?

Reply:
If it is a discriminator parameter of base class, the `DiscriminatorValue` could be null. But if the model is the derived model,  it is not possible that the `IsDiscriminator` be true but `DiscriminatorValue` is null(in real word), but in our model code there is no any constrain to persist this.